State initial_state
---
int return_message
